### DISCUSSION TITLE
Is there a 'Lexical way' to copy text content to clipboard ?

### DISCUSSION THREAD

**Discussion by aaronzomback** [facebook/lexical#4657] [Q&A]
I'm currently trying to create a copy to clipboard function, where I have a clipboard icon above the editor.

Here is my current implementation:
```tsx
  const stringifiedEditorState = editor?.getEditorState()
  const parsedEditorState = stringifiedEditorState && editor?.parseEditorState(stringifiedEditorState.toJSON())
  const captionPreview = parsedEditorState?.read(() => $getRoot().getTextContent())
  const onCopyToClipboard = (e: React.MouseEvent<HTMLElement>) => {
    e.stopPropagation()
    setTextCopied(true)
    navigator.clipboard.writeText(captionPreview ?? '')
    setTimeout(() => setTextCopied(false), 1500)
  }
  ```
  
 I am able to receive the copied content, but for some reason it seems to enter 1 more line than between new paragraph nodes.
  
 I took a look at the `copyToClipboard `plugin, but I don't understand how the API is supposed to work or help.
  
 Any insights would be greatly appreciated.
 
 ❗️I can also confirm it seems to be registering linebreaks for each paragraph node. But it also registers multiple sometimes as confirmed by the `/n` line breaks in the console log
 
![Screenshot 2023-06-14 at 16 05 27](https://github.com/facebook/lexical/assets/63470294/94f49e30-d6dd-4b7a-ab77-7bde97eeb1ea)
![Screenshot 2023-06-14 at 16 05 16](https://github.com/facebook/lexical/assets/63470294/4efda0d1-0f6a-42f4-8921-5fa0c05e3a8b)

 
![Screenshot 2023-06-14 at 15 12 38](https://github.com/facebook/lexical/assets/63470294/de64a858-3c18-4f09-9be4-abd67972f421)
![Screenshot 2023-06-14 at 15 12 14](https://github.com/facebook/lexical/assets/63470294/50b5610b-34e4-48ff-bcc0-b80ef56aa229)

**Comment by ivailop7** [comment]
There isn't an 'officially endorsed way', but a slightly better implementation you can have a look is this one: https://github.com/facebook/lexical/blob/5162dd2d7e139d4c489601c80d7abe5d40211f1d/packages/lexical-playground/src/nodes/TableComponent.tsx#L1139-L1168

**Comment by aaronzomback** [comment]
Thanks @ivailop7 for your reply -- 

I actually think to solve the problem I'm facing I need to override `ParagraphNodes` with `LineBreakNodes` since the `ParagraphNode` contains too many `\n` line break characters causing too much space between lines when copy + pasting the text content into social media posts.

I'm not sure if this is the correct approach, since it seems initially a `ParagraphNode` must be appended to the `RootNode` in order for text content to be inserted.

From your experience, do you have any better ideas how this can be achieved ?

Thanks a lot
